### PR TITLE
Fix off by one error in concreteSeriesSet

### PR DESCRIPTION
**Pull request by brian-brazil**
This was losing us the first series from remote read.

@fabxc

### CODE REVIEW

File: storage/remote/read.go [hunk 1]
```diff
@@ -151,11 +151,11 @@ type concreteSeriesSet struct {
 
 func (c *concreteSeriesSet) Next() bool {
 	c.cur++
-	return c.cur < len(c.series)
+	return c.cur-1 < len(c.series)
 }
 
 func (c *concreteSeriesSet) At() storage.Series {
-	return c.series[c.cur]
+	return c.series[c.cur-1]
```
Comment (grobie) [inline storage/remote/read.go]: Intuitively I wouldn't expect `At` to return the series before the current position. Could you extend the test to include at least 2 series and show the behavior of Next and At there?

Comment (brian-brazil) [reply storage/remote/read.go]: Done

Comment (Gouthamve) [reply storage/remote/read.go]: Yep, a little counter-intuitive. This is what we do in tsdb: https://github.com/prometheus/tsdb/blob/dc8710380793a6d30b94e730edf6c8a4dae69945/postings.go#L319-L364

Comment (grobie) [reply storage/remote/read.go]: Thanks @brian-brazil.

**Pull request merged**